### ISSUE TITLE
ERROR: A package archive must be named <pack>-<version>.<ext>

### ISSUE THREAD

**Issue by lancejpollard** [SWI-Prolog/swipl-devel#11]
Hello,

I am new to prolog and am trying to learn to create prolog packages. I have created a GitHub repo here:

https://github.com/prolibs/app

I then have tried several ways of creating releases here:

https://github.com/prolibs/app/releases

For each release, I have tried taking the `.zip` URL and passing it to `pack_install`, but it always errors out. Here is an example:

``` prolog
?- pack_install('https://github.com/prolibs/app/archive/0.1.1.zip').
ERROR: 0.1.1.zip: A package archive must be named <pack>-<version>.<ext>
% Contacting server at http://www.swi-prolog.org/pack/query ... done
Warning: No registered pack matches "https://github.com/prolibs/app/archive/0.1.1.zip"
false.
?-
```

In that example, the name of the GitHub release is `0.1.1`, and so it creates the file `https://github.com/prolibs/app/archive/0.1.1.zip`.

I have also tried naming the release `app-0.1.1`, so the URL becomes `https://github.com/prolibs/app/archive/app-0.1.1.zip`, but that also results in an error:

``` prolog
?- pack_install('https://github.com/prolibs/app/archive/app-0.1.1.zip').
%  library(ssl) compiled into ssl 0.00 sec, 43 clauses
%    library(rbtrees) compiled into rbtrees 0.01 sec, 190 clauses
%   library(thread_pool) compiled into thread_pool 0.01 sec, 238 clauses
%   http_wrapper compiled into httpd_wrapper 0.00 sec, 68 clauses
%  thread_httpd compiled into thread_httpd 0.02 sec, 407 clauses
% library(http/http_ssl_plugin) compiled into http_ssl_plugin 0.02 sec, 462 clauses
ERROR: app-app-0.1.1: A package archive must be named <pack>-<version>.<ext>
ERROR: pack_file `pack.pl' does not exist
?-
```

The problem there is, if you name your GitHub release something like `<name>-<version>` (which they recommend not to do anyways), then when the `.zip` file gets unpacked, it results in a folder called `<name>-<name>-version`, which is the reason for `ERROR: app-app-0.1.1`.

So basically, you can't use GitHub's release feature to host prolog packages, which is kind of a bummer. It seems like something that would be straightforward to fix. Is there anything that can be done to support this? If not, what is your recommended workaround?

Thanks for your help.
